plugins/woocommerce-admin/client/marketing/overview-multichannel/Campaigns/Campaigns.test.tsx

```diff
@@ -131,4 +155,30 @@ describe( 'Campaigns component', () => {
 		// Campaign info in the second page.
 		expect( screen.getByText( 'Campaign 6' ) ).toBeInTheDocument();
 	} );
+
+	it( 'renders a "Create new campaign" button in the card header, and upon clicking, displays the "Create a new campaign" modal', async () => {
+		( useCampaigns as jest.Mock ).mockReturnValue( {
+			loading: false,
+			error: undefined,
+			data: [ createTestCampaign( '1' ) ],
+			meta: {
+				total: 1,
+			},
+		} );
+		( useNewCampaignTypes as jest.Mock ).mockReturnValue( {
+			loading: false,
+		} );
+
+		render( <Campaigns /> );
+
+		expect( screen.getByText( 'Create new campaign' ) ).toBeInTheDocument();
+
+		await userEvent.click(
+			screen.getByRole( 'button', { name: 'Create new campaign' } )
+		);
```

Comment:
How about using `getByRole( 'button', ...` instead of `screen.getByText( 'Create new campaign' )` ?

```suggestion
		const button = screen.getByRole( 'button', {
			name: /create new campaign/i,
		} );

		expect( button ).toBeInTheDocument();

		await userEvent.click( button );
```

Reply:
Ah yes, we should always use `getByRole` over `getByText` when possible, based on https://testing-library.com/docs/queries/about/#priority. I changed one line and missed the other line. 

Looking at it now, we actually don't need the `expect( screen.getByText( 'Create new campaign' ) ).toBeInTheDocument();` at all. When we do `getByRole`, if the element is not found, it will throw an error, so I have removed the line in [`551db21` (#37044)](https://github.com/woocommerce/woocommerce/pull/37044/commits/551db21e9bfd1155afd8cbb64b70341ee3c6345f).